tests/testthat/test-read-csv.R

```diff
@@ -0,0 +1,37 @@
+context("read_csv parameter testing")
+
+#Formatting for column imputation, col_name detection and NA detection.
+test_data <- read_csv(file.path(getwd(),"/basic-df.csv"), col_types = NULL, col_names = TRUE)
+
+test_that("read_csv automatic column type imputation works",{
+  expect_that(unname(unlist(lapply(test_data,class))), equals(c("logical","integer","numeric","character")))
+})
+
+test_that("read_csv's col_names = TRUE actually picks up col_names",{
+  expect_that(names(test_data), equals(c("a","b","c","d")))
```

Comment:
At least on the Ruby side, I hear general TDD practice is to put one expectation per test, so that test failures narrow down to a single assertion and minimize further debugging. What do you think of

```r
describe("using read_csv on basic-df.csv", {
   # code from above
})